appserver/distributions/glassfish-common/src/main/resources/bin/stopserv

```diff
@@ -15,7 +16,16 @@
 # SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
 #
 
-AS_INSTALL=`dirname "$0"`/../glassfish
-AS_INSTALL_LIB="$AS_INSTALL/modules"
+export AS_INSTALL="$(dirname "$(realpath -s "$0")")/../glassfish"
 
-exec java -jar "$AS_INSTALL_LIB/admin-cli.jar" stop-domain "$@"
+case "`uname`" in
+  CYGWIN*) AS_INSTALL=`cygpath --windows $AS_INSTALL`
+esac
+. "${AS_INSTALL}/config/asenv.conf"
+
+JAVA=java
+if [ ${AS_JAVA} ]; then
+    JAVA=${AS_JAVA}/bin/java
```

Comment:
Should be in quotes in case contains spaces:

```suggestion
    JAVA="${AS_JAVA}"/bin/java
```

Reply:
Not needed here, but it is needed below when using exec, if I remember well.